doc/source/glossary.rst

```diff
@@ -373,24 +364,6 @@ Glossary
        immutable.  Two common examples are strings and tuples.
```

Comment:
Immutable is also in the python glossary, with a longer description

Reply:
My glossary doesn't include this. As I understand it, only the '+' in this PR are up for review; these denote wholly new entries. Deleted and revised entries appear as diffs in two other PRs.

Reply:
Which are those PRs?

Reply:
Deletions was #17264.

For revised terms, see @mattip's comment: https://github.com/numpy/numpy/pull/17263#issuecomment-689575679

Reply:
Are you suggesting we open a new PR for further deletions?